students/stellie/lesson09/donor_models.py

```diff
@@ -52,15 +51,12 @@ def add_new_donor(self, donor_name, donation_amount):
 
     # Create report for user to see list of all donors and donations made
     def create_report(self):
-        def sum_total(donor_record):
-            return(sum(donor_record[1]))  # return donor donations sum in DB
-        donor_stat = []
         # Sort database by sum amounts in descending order
-        sorted_db = sorted(self.items, key=sum_total, reverse=True)
+        sorted_db = sorted(self.data.items(), key=lambda donor_record:
+                           sum(donor_record[1]), reverse=True)
         for item in sorted_db:
-            total = Donor(item[0], item[1]).total_donations()
-            count = Donor(item[0], item[1]).donation_count()
-            average = Donor(item[0], item[1]).avg_donations()
-            donor_stat.append(f'{item[0]:<20} | {total:<12.2f} | '
-                              f'{count:<10} | {average:<15.2f}')
-        return donor_stat
+            donor_record = Donor(item[0], item[1])
+            yield (f'{donor_record.name:<20} | '
+                   f'{donor_record.total_donations():<12.2f} | '
+                   f'{donor_record.donation_count():<10} | '
+                   f'{donor_record.avg_donations():<15.2f}')
```

Comment:
Hi @natasha-aleksandrova thanks for the example.  I thought to put something in the Donor class so that I wouldn't have to recreate objects but thought it'd require a lot more refactoring.  I appreciate you taking a second look though.  I have completed this course but will probably tinker with this assignment for fun and redo the assignment for practice!

Reply:
@stellie82 yes you have completed and passed and not required to make any additional changes 🙂 

I wanted to make sure that I point this out anyway as I'm sure you would want to know for your learning benefit. Good luck in the next course!